src/hotspot/share/cds/cdsEnumKlass.cpp

```diff
@@ -89,10 +89,11 @@ void CDSEnumKlass::archive_static_field(int level, KlassSubGraphInfo* subgraph_i
               ik->external_name(), fd.name()->as_C_string());
   }
   oop oop_field = mirror->obj_field(fd.offset());
+  Klass* oop_field_klass = oop_field->is_objArray() ? oop_field->klass()->super() : oop_field->klass();
```

Comment:
Hmm, what does this do? I assume `is_objArray` tests if something can be converted to Java `Object[]`. In this case, CDS should check if the component type of the array instead of the array supepr class, I think? The original code did not call `super()`.